src/api/middleware/validators/company.js

```diff
@@ -55,18 +55,35 @@ const block = useExpressValidators([
         .trim(),
 ]);
 
-const unblock = useExpressValidators([
+const disable = useExpressValidators([
+    param("companyId")
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND),
+    body("adminReason")
+        .if((reason, { req }) => req.hasAdminPrivileges).exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .isString().withMessage(ValidationReasons.STRING).bail()
+        .trim(),
+]);
+
+const unhide = useExpressValidators([
     param("companyId")
         .exists().withMessage(ValidationReasons.REQUIRED).bail()
         .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
         .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND),
 ]);
 
+/* Since their logic is the same, create a common validator and export it with a more meaningfull name.
+ * Not done with block/disable because the validation chain is slightly different
+ */
 module.exports = {
     finish,
     list,
     block,
-    unblock,
+    unhide,
```

Comment:
I think we should just normalize it as having two different exports for the same thing will cause confusion in the future. Just call it enable and use for unblocking and unhiding

```suggestion
    enable,
```